### PR TITLE
Update logging

### PR DESCRIPTION
**Pull request by pnuu**
This PR does a "full" overhaul for the logging in Trollflow2:
- [x] pass the logging config also to the sub-process
- [x] harmonize logger variable naming
- [x] adjust log levels
- [x] make sure the log messages are meaningful and compact
- [x] add a full example of logging config to documentation

Also, `isort` plugin was added to pre-commit so that the imports are properly grouped and sorted. All the complaints from `pre-commit run --all-files` were fixed.

 - [x] Tests added  
 - [x] Tests passed 
 - [x] Passes ``flake8 trollflow2`` 
 - [x] Fully documented

### CODE REVIEW

**Comment by pnuu** [comment]
Documentation is still to be updated, but in the meanwhile I'll mark this as ready for review.

And for reference, below you can see an example log output from my MTG/FCI processing test.

Log config:
```yaml
version: 1
formatters:
  fmt:
    format: '[%(asctime)s %(levelname)-8s %(name)s] %(message)s'
handlers:
  console:
    class: logging.StreamHandler
    formatter: fmt
    stream: ext://sys.stdout
loggers:
  '':
    level: WARNING
    handlers: [console]
  'trollflow2':
    level: INFO
    handlers: [console]
    propagate: false
```

Log output (the publish rows are truncated for clarity) for two consecutive time slots:
```
[2023-02-01 13:05:37,248 INFO     trollflow2.launcher] Launching Satpy-based runner.
[2023-02-01 13:05:37,252 INFO     trollflow2.launcher] Launching trollflow2 with subprocesses
[2023-02-01 13:09:02,779 INFO     trollflow2.launcher] New message received.
[2023-02-01 13:09:04,006 INFO     trollflow2.plugins] Creating scene
[2023-02-01 13:09:40,783 INFO     trollflow2.plugins] Loading 1 composites.
[2023-02-01 13:09:50,079 INFO     trollflow2.plugins] Resampling to EPSG_3035_2km
/opt/conda/lib/python3.10/site-packages/dask/core.py:119: RuntimeWarning: invalid value encountered in log
  return func(*(_execute_task(a, cache) for a in args))
/opt/conda/lib/python3.10/site-packages/dask/array/chunk.py:276: RuntimeWarning: invalid value encountered in cast
  return x.astype(astype_dtype, **kwargs)
[2023-02-01 13:09:54,039 INFO     trollflow2.plugins.s3] Uploading data to S3.
[2023-02-01 13:09:54,231 INFO     trollflow2.plugins] Publishing pytroll://mtg/0deg/fci/image/EPSG3035 file ...
[2023-02-01 13:09:55,188 INFO     trollflow2.launcher] All 1 files produced nominally in 0:00:52.198179
[2023-02-01 13:19:02,742 INFO     trollflow2.launcher] New message received.
[2023-02-01 13:19:04,218 INFO     trollflow2.plugins] Creating scene
[2023-02-01 13:19:41,990 INFO     trollflow2.plugins] Loading 1 composites.
[2023-02-01 13:19:52,910 INFO     trollflow2.plugins] Resampling to EPSG_3035_2km
/opt/conda/lib/python3.10/site-packages/dask/core.py:119: RuntimeWarning: invalid value encountered in log
  return func(*(_execute_task(a, cache) for a in args))
/opt/conda/lib/python3.10/site-packages/dask/array/chunk.py:276: RuntimeWarning: invalid value encountered in cast
  return x.astype(astype_dtype, **kwargs)
[2023-02-01 13:19:56,758 INFO     trollflow2.plugins.s3] Uploading data to S3.
[2023-02-01 13:19:56,997 INFO     trollflow2.plugins] Publishing pytroll://mtg/0deg/fci/image/EPSG3035 file ...
[2023-02-01 13:19:57,865 INFO     trollflow2.launcher] All 1 files produced nominally in 0:00:55.099049
```

**Comment by pnuu** [comment]
The first update already in, I'll make the `Launching trollflow2 with subprocesses` a debug message.

**Comment by pnuu** [comment]
And this relevant bit of info was missing above, the plugins I'm using:
```yaml
workers:
  - fun: !!python/name:trollflow2.plugins.create_scene
  - fun: !!python/name:trollflow2.plugins.load_composites
  - fun: !!python/name:trollflow2.plugins.resample
  - fun: !!python/name:trollflow2.plugins.save_datasets
  - fun: !!python/name:trollflow2.plugins.s3.uploader
  - fun: !!python/object:trollflow2.plugins.FilePublisher {nameservers: False, port: 40000}
  ```

**Comment by pnuu** [comment]
The documentation is now split into separate pages. Especially the `Launcher` page will need quite a lot of attention to be useful. Also `Product list` should be clarified. But these are tasks for other PRs.

File: doc/source/logging.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+Finer control for different loggers (packages and their modules) can be controlled by adding a new section under
+``loggers``. For example getting DEBUG level (and above) messages from Pyspectral, this can be used:
+
+.. code-block:: yaml
+
+    'pyspectral':
+      level: DEBUG
+      handlers: [console_handler]
+      propagate: false
```
Comment (mraspaud) [inline doc/source/logging.rst:43]: I think this needs to be removed
```suggestion
```

File: doc/source/logging.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      backupCount: 14
+      utc: true
+  loggers:
+    '':
+      level: WARNING
+      handlers: [console]
+    'trollflow2':
+      level: INFO
+      handlers: [console]
+      propagate: false
```
Comment (mraspaud) [inline doc/source/logging.rst:77]: This needs to be removed also
```suggestion
```

File: trollflow2/tests/test_logging.py [hunk 3]
```diff
@@ -76,36 +117,89 @@ def test_log_config_is_used_when_provided():
...
+        logger = logging.getLogger(log_name)
+        logger.debug(f"{log_name} debug")
+        logger.info(f"{log_name} info")
+        logger.warning(f"{log_name} warning")
+
+
 @pytest.mark.skipif(sys.platform != "linux",
                     reason="Logging from a subprocess seems to work only on Linux")
-def test_logging_works_in_subprocess(caplog):
+def test_logging_works_in_subprocess_with_default_logging_config(caplog):
```
Comment (mraspaud) [inline trollflow2/tests/test_logging.py:151]: This is actually the same test as `test_queued_logging_process_default_config`, I think it can be removed.

**Review by mraspaud — COMMENTED** [review]
this LGTM, but I might be partial... some comments inline.

File: trollflow2/tests/test_logging.py (hunk 3 above)
Comment (pnuu) [reply trollflow2/tests/test_logging.py]: I merged the two, `test_queued_logging_process_default_config` didn't have the duplicate lines check, so I added that.

**Pull request merged**